src/install-script/installer.ts

```diff
@@ -0,0 +1,200 @@
+import * as archiver from "archiver";
+import { ChildProcess, exec, spawn } from "child_process";
+import { createWriteStream, ensureDir, remove } from "fs-extra";
+import { join as joinPath } from "path";
+
+export enum ScriptMessageType {
+  MINOR,
+  MAJOR,
+  ERROR,
+  COMPLETE
+}
+
+export enum InfoSource {
+  NPM = "npm",
+  VERDACCIO = "verdaccio",
+  OFFLINE_NPM = "offline-npm",
+  ARCHIVER = "archiver"
+}
+
+export interface IScriptMessage {
+  type: ScriptMessageType;
+  message: string;
+  source: InfoSource;
+}
+
+// Utility to make using callbacks with IScriptMessages easier
+const makeOutputter = (
+  callback: (data: IScriptMessage) => void,
+  type: ScriptMessageType
+) => (message: string, source: InfoSource = InfoSource.OFFLINE_NPM) => {
+  callback({
+    message,
+    source,
+    type
+  });
+};
+
+/**
+ * Installs an npm cache through Verdaccio and bundles the results in a zip file.
+ * @param packages the list of package names to install
+ * @param onInfo callback function for info messages
+ * @param onError callback function for error messages
+ * @param onComplete callback function for when the installation is complete
+ * @param outputPath the directory in which storage.zip should be placed
+ * @param verdaccioPath the directory in which the verdaccio source is located (usually node_modules)
+ */
+export const installCache = (
+  packages: string[],
+  onInfo: (data: IScriptMessage) => void,
+  onError: (data: IScriptMessage) => void,
+  onComplete: (data: IScriptMessage) => void,
```

Comment:
It'd be cleaner to have a single method passed in that takes in the data, and add a "severity"-like enum (info/error/complete) to that. Then, instead of passing `print, print, print` to this method, you can handle (or choose not to handle) the level elsewhere.

Maybe `ScriptMessageType` can already do that for you?